include/SimulationManager.h

```diff
@@ -0,0 +1,34 @@
+
+#ifndef REST_SIMULATIONMANAGER_H
+#define REST_SIMULATIONMANAGER_H
+
+#include <TRestGeant4Event.h>
+#include <TRestGeant4Metadata.h>
+#include <TRestGeant4PhysicsLists.h>
+#include <TRestGeant4Track.h>
+#include <TRestRun.h>
+
+class SimulationManager {
+   public:
+    SimulationManager();
+    ~SimulationManager();
+
+    TRestRun* fRestRun = nullptr;
+    TRestGeant4Track* fRestGeant4Track = nullptr;
+    TRestGeant4Event *fRestGeant4Event = nullptr, *fRestGeant4SubEvent = nullptr;
+    TRestGeant4Metadata* fRestGeant4Metadata = nullptr;
+    TRestGeant4PhysicsLists* fRestGeant4PhysicsLists = nullptr;
+    Int_t fBiasing = 0;
+
+    static const Int_t maxBiasingVolumes = 50;
+
+    // These histograms would be better placed inside TRestGeant4BiasingVolume
+    TH1D* biasingSpectrum[maxBiasingVolumes];
+    TH1D* angularDistribution[maxBiasingVolumes];
+    TH2D* spatialDistribution[maxBiasingVolumes];
```

Comment:
I believe would be better to use `std::vector` instead of c arrays

Reply:
yes, this will be deprecated in the future (if @jgalan agrees) because currently its really messy and I don't really know if it even works, anyone is invited to refactor this instead.